kernel/src/test/java/examples/DeadlockDocTest.java

```diff
@@ -22,57 +22,51 @@
  */
 package examples;
 
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
 import java.util.concurrent.TimeUnit;
 
-import org.neo4j.doc.test.rule.EmbeddedDatabaseRule;
 import org.neo4j.graphdb.GraphDatabaseService;
 import org.neo4j.graphdb.Transaction;
 import org.neo4j.graphdb.TransactionFailureException;
+import org.neo4j.harness.junit.extension.Neo4jExtension;
 import org.neo4j.internal.helpers.TransactionTemplate;
 import org.neo4j.kernel.DeadlockDetectedException;
 
-public class DeadlockDocTest
+@ExtendWith( Neo4jExtension.class )
+class DeadlockDocTest
 {
-    @Rule
-    public final EmbeddedDatabaseRule rule = new EmbeddedDatabaseRule();
-
     @Test
-    public void transactionWithRetries() throws InterruptedException
+    void transactionWithRetries( GraphDatabaseService databaseService )
     {
-        Object result = transactionWithRetry();
+        Object result = transactionWithRetry( databaseService );
     }
 
     @Test
-    public void transactionWithTemplate() throws InterruptedException
+    void transactionWithTemplate( GraphDatabaseService databaseService )
     {
-        GraphDatabaseService graphDatabaseService = rule.getGraphDatabaseAPI();
-
         // tag::template[]
         TransactionTemplate template = new TransactionTemplate(  ).retries( 5 ).backoff( 3, TimeUnit.SECONDS );
```

Comment:
This was moved to `*.internal.helpers.*`. Let's make a note to add it back or remove this snippet altogether.

Reply:
yep yep, ideally when we will enable our super check that will check that this repo uses only public api we will remove everything that is not part of that :) (in the bright future)